src/systemdhook.c

```diff
@@ -999,19 +999,20 @@ int main(int argc, char *argv[])
 	}
 #endif
 
-	/* OCI hooks set target_pid to 0 on poststop, as the container process already
-	   exited.  If target_pid is bigger than 0 then it is the prestart hook.  */
-	if ((argc > 2 && !strcmp("prestart", argv[1])) || target_pid) {
+	if (argc > 2 && !strcmp("prestart", argv[1])) {
 		if (prestart(rootfs, id, target_pid, mount_label, config_mounts, config_mounts_len, uid, gid) != 0) {
 			return EXIT_FAILURE;
 		}
-	} else if ((argc > 2 && !strcmp("poststop", argv[1])) || (target_pid == 0)) {
+	} else if (argc > 2 && !strcmp("poststop", argv[1])) {
```

Comment:
doesn't PostStart always specify `target_pid != 0`?  From my understanding, it is 0 only when the container is already terminated so there is no process left.  Either way, having it only for argc==1 sounds good to me as well